File: 0002-add-two-numbers/0002-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    dummy := &ListNode{Val: 0}
    cur := dummy
    extra := 0
    for l1 != nil || l2 != nil || extra != 0 {
        if l1 != nil {
            extra += l1.Val
            l1 = l1.Next
        }
        if l2 != nil {
            extra += l2.Val
            l2 = l2.Next
        }
        cur.Next = &ListNode{Val: extra % 10}
        cur = cur.Next
        extra = extra / 10

    }
    return dummy.Next
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        extra = 0
        dummy = ListNode(0)
        cur = dummy
        while l1 or l2 or extra != 0:
            ans = extra
            if l1:
                ans += l1.val
                l1 = l1.next
            if l2:
                ans += l2.val
                l2 = l2.next
            cur.next = ListNode(ans%10)
            cur = cur.next
            extra = ans//10
        return dummy.next